app/models/akka/SubscriptionChannel.scala

```diff
@@ -0,0 +1,107 @@
+package models.akka
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.stream._
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler, StageLogging}
+import models.rpc.{DSAMessage, ResponseMessage, SubscriptionNotificationMessage}
+import akka.pattern.ask
+import akka.util.Timeout
+import models.akka.Messages._
+
+import scala.concurrent.duration._
+import scala.collection.mutable
+import scala.concurrent.{Await, ExecutionContext}
+
+
+class SubscriptionChannel(val store: ActorRef)
+                         (implicit actorSystem: ActorSystem, materializer: Materializer)
+  extends GraphStage[FlowShape[SubscriptionNotificationMessage, DSAMessage]] {
+
+  type Sid = Int
+
+  val in = Inlet[SubscriptionNotificationMessage]("Subscriptions.in")
+  val out = Outlet[DSAMessage]("Subscriptions.out")
+  implicit val as = actorSystem
+  implicit val m = materializer
+  implicit val ctx = ExecutionContext.global
```

Comment:
Probably better either import it as implicit, or just `import actorSystem.dispatcher`

Reply:
Done